test/src/test/java/jenkins/security/Security2779Test.java

```diff
@@ -47,7 +42,8 @@ public void noXssInHelp() throws Exception {
         assertThat(eventResult, instanceOf(boolean.class));
         Assert.assertTrue((boolean) eventResult);
         webClient.waitForBackgroundJavaScript(2000);
-        Assert.assertEquals(0, alerts.get());
+        // Assertion includes the selector for easier diagnosis
```

Comment:
Well, better still would be the text of the alert! But that only matters if the test fails, and we hope no refactoring will ever make it fail again…right?

Reply:
Except when it does: https://github.com/jenkinsci/jenkins/pull/7134/checks?check_run_id=8465259022

Reply:
I'm not sure how the text of the alert would help in this case, unless the alert content were changed.  Right now, the alert text has the value ('1').  It could be changed to state the selector that is being tested, but since that is already in the test, I thought it would not help much to get that message from the alert.

Have I missed something in my understanding?

Reply:
Right, never mind.